### DISCUSSION TITLE
Validating Invoice is not working.

### DISCUSSION THREAD

**Discussion by bm438** [stephanstapel/ZUGFeRD-csharp#479] [General]
Hello,

I am using this to try to validate, how is that done, if I load any xml it always go though, even with bullshit ones:
`public static bool IsValidXInvoiceXML(string filePath)
{
    bool result = false;
 
    try
    {
        InvoiceDescriptor descriptor = InvoiceDescriptor.Load(filePath);
 
        switch (descriptor.Profile.ToString())
        {
            case "Basic":
            case "Comfort":
            case "Extended":
            case "Minimum":
            case "BasicWL":
            case "EReporting":
            case "Unknown":
                result = false;
                break;
 
            case "XRechnung":
            case "XRechnung1":
                result = true;
                break;
 
            default:
                result = false;
                break;
        }
    }
    catch(Exception ex)
    {
        Log.Exception(ex);
        return false;
    }
 
    return result;
}`

I get eveverytime a Profile, not matter what I delete from the XML, am I doing it wrong?

I have put my broken xml which I created into here, an online Validator also trows error like it should.

[Basic_XML_Broken 1.txt](https://github.com/user-attachments/files/18019813/Basic_XML_Broken.1.txt)

**Comment by stephanstapel** [comment]
Would you mind making this a test case? The sample file should be in the repository, so loading it and doing this basic check should be easy. I will then take care about the rest.

**Comment by bm438** [comment]
@stephanstapel I can do that, but just for clarification, this lib has not capability of validating if an xml from an X-Rechnung or Zugpferd is valid or has the valid fields or an schema?

**Comment by stephanstapel** [comment]
I, now I understand. Yes, you are right. There is currently no validation implemented in this component. 
There is an ongoing discussion here:
https://github.com/stephanstapel/ZUGFeRD-csharp/discussions/316

and there is some code in another component that could be ported but that is none of my priorities currently.
